Test unversioned base URLs

CasperWA opened the pull request:
Closes #31 

Change the validator action to check the mandatory base URLs:
- Unversioned base URL.
- Major-versioned base URL.

The input `all versioned paths` will now simply add validation for the Major.Minor-versioned base URL as well as the Major.Minor.Patch-versioned base URL.

The input `path` MUST now point to the unversioned OPTIMADE base URL.

CasperWA commented:
@ml-evs This should be ready for review as well :)

ml-evs commented on README.md:
```diff
@@ -42,16 +42,27 @@ with:
 | `fail_fast` | Whether or not to exit and return a non-zero error code on first failure. | Optional | `false` | `v2+`
 | `index` | Whether or not this is an index meta-database | Optional | `false` | `v1+`
-| `path` | Path for the OPTIMADE (versioned) base URL - MUST start with `/`<br>_Note_: If `all versioned paths` is `true`, this MUST be un-versioned, e.g., `/optimade` or `/`, otherwise it MUST be versioned, e.g., the default value | Optional | `/v1` | `v1+`
+| `path` | Path for the OPTIMADE (versioned) base URL - MUST start with `/`<br>The path MUST NOT include the versioned part of the base URL. Rather, it MUST point to the unversioned base URL of your OPTIMADE implementation. | Optional | `/` | `v1+`
```
Could you clarify this a bit more? I know what you mean, but now the example value has changed to just `"/"` it might be unclear:

```suggestion
| `path` | Path to append to the domain to reach the OPTIMADE unversioned base URL - MUST start with `/`<br>The path MUST NOT include the versioned part of the base URL. Rather, it MUST point to the unversioned base URL of your OPTIMADE implementation` | Optional | `/` | `v1+`
```

Does this setting still make sense now or could all this info be bundled into the domain?

CasperWA replied:
Since the domain does not include the path, it cannot be added to the domain. Or do you mean to add `/` to the domain if not already present? That might be a great idea.

I put in all these `MUST BE /` though, to avoid having to do a million checks for prep- or appended `/`.

ml-evs commented on action.yml:
```diff
@@ -33,21 +32,24 @@ inputs:
 
   path:
-    description: Path of the OPTIMADE (versioned) base URL - MUST start with '/'
+    description: >
```
Guess this desc would need to be updated following my comment above too

ml-evs reviewed (commented):
Hi @CasperWA, just a couple of minor comment below, otherwise looks good.

Extra thought: currently the validator-action v2 doesn't work with any python-tools version <0.10 as we changed from `optimade_validator` to `optimade-validator` and updated the flags. 

We should either mention this in the README and hardcode a failure with a useful message if the requested validator version is too low, or change the way the validator is called for versions <0.10. I can raise an issue about this after lunch.

CasperWA commented:
> Extra thought: currently the validator-action v2 doesn't work with any python-tools version <0.10 as we changed from `optimade_validator` to `optimade-validator` and updated the flags.
> 
> We should either mention this in the README and hardcode a failure with a useful message if the requested validator version is too low, or change the way the validator is called for versions <0.10. I can raise an issue about this after lunch.

Good point!
I believe this is already mentioned in the README (see the bottom), but if the fix is "easy", we should definitely consider adding support for older OPTIMADE Python tools versions for v2 as well. Please do raise the issue when you have time 👍

CasperWA commented on tests/test_fixtures.bash:
```diff
@@ -6,29 +6,47 @@ if [ -z "${DOCKER_BATS_WORKDIR}" ]; then
+    export TEST_MAJOR_RUN_VALIDATOR="${TEST_BASE_RUN_VALIDATOR}v1"
 
+    # Clean "cache"
     rm -f ${DOCKER_BATS_WORKDIR}/tests/.entrypoint-run_validator.txt
+    rm -f ${ENTRYPOINT_SH}
```
`teardown()` and `teardown_file()` are automatically called after every single test and every single test file, respectively, so no need to call them here, however, they could be called here as well I guess?
The current way is more explicit, but is don't mind changing it to save on text... :)

ml-evs commented:
> > Extra thought: currently the validator-action v2 doesn't work with any python-tools version <0.10 as we changed from `optimade_validator` to `optimade-validator` and updated the flags.
> > We should either mention this in the README and hardcode a failure with a useful message if the requested validator version is too low, or change the way the validator is called for versions <0.10. I can raise an issue about this after lunch.
> 
> Good point!
> I believe this is already mentioned in the README (see the bottom), but if the fix is "easy", we should definitely consider adding support for older OPTIMADE Python tools versions for v2 as well. Please do raise the issue when you have time +1

While the fix would be easy enough, i.e. just link optimade_validator to the new optimade-validator and make sure the flags that are passed are compatible, I'm not sure it's worth it... as mentioned in my issue #35 I think we should just raise an error for <0.10.

ml-evs reviewed (approved):
Happy with this when you are @CasperWA, thanks

CasperWA commented:
@ml-evs Still interested to hear what you exactly meant about the domain+path. Did you mean to put the `/` into domain and try to be less stringent in the values of both domain and path, i.e., implementing a lof of checks for prep-/appended `/`?
